src/main/java/org/jenkinsci/plugins/docker/commons/tools/DockerToolInstaller.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2015 Jesse Glick, CloudBees Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.docker.commons.tools;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.ProxyConfiguration;
+import hudson.model.Node;
+import hudson.model.TaskListener;
+import hudson.remoting.Callable;
+import hudson.remoting.VirtualChannel;
+import hudson.tools.ToolInstallation;
+import hudson.tools.ToolInstaller;
+import hudson.tools.ToolInstallerDescriptor;
+import hudson.util.IOUtils;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLConnection;
+
+
+/**
+ * Download and install Docker CLI binary, see https://docs.docker.com/installation/binaries/
+ * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
+ */
+public class DockerToolInstaller extends ToolInstaller {
+
+    private final String version;
+
+    @DataBoundConstructor
+    public DockerToolInstaller(String label, String version) {
+        super(label);
+        this.version = version;
+    }
+
+    public String getVersion() {
+        return version;
+    }
+
+    @Override
+    public FilePath performInstallation(ToolInstallation toolInstallation, Node node, TaskListener listener) throws IOException, InterruptedException {
+
+        String os = node.getChannel().call(new Callable<String, IOException>() {
+            public String call() throws IOException {
+                String os = System.getProperty("os.name").toLowerCase();
+                String arch = System.getProperty("os.arch").contains("64") ? "x86_64" : "i386";
```

Comment:
samples I've seen so far do report x86_64 or amd64.
This could be improved later, especially to support arm targets